ratpack-kafka-producer/src/test/groovy/smartthings/ratpack/kafka/KafkaProducerServiceSpec.groovy

```diff
@@ -156,4 +156,43 @@ class KafkaProducerServiceSpec extends Specification {
 		assert result.throwable.getClass() == IllegalStateException
 		assert result.throwable.getMessage() == 'KafkaProducer is currently not available.'
 	}
+
+	def "Can send a message with optional properties"() {
```

Comment:
Because kafka is ultimately using the properties it might be good to test that the properties generated by this config are what you expect.  So that things are set properly if set, and that they are not set if unset.  Maybe just 2 tests one that checks when everything is set, and one that tests when nothing optional is set.

Having this more functional like test is also probably good to keep as well, but the above until test will check that things going to kafka are what you expect.